### PR TITLE
admin tools for PromotedAddon|Approval

### LINKED ISSUE #14861: Add admin tooling to manage Promoted addons
> Admin tool to manage the lists of add-ons. Any add-ons (regardless of type, status, or distribution method) can be added to or removed from these lists.

> An add-on can only belong to a single list at any time.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #14861

### CODE REVIEW

File: src/olympia/promoted/admin.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+class PromotedApprovalInlineChecks(admin.checks.InlineModelAdminChecks):
+    def _check_relation(self, obj, parent_model):
+        """PromotedApproval doesn't have a direct FK to PromotedAddon (it's via
+        Addon, Version) so we have to bypass this check.
+        """
+        return []
+
+
+class PromotedApprovalInline(admin.TabularInline):
+    model = PromotedApproval
```
Comment (eviljeff) [inline src/olympia/promoted/admin.py:20]: @bobsilverberg this is for PromotedApproval

Comment (bobsilverberg) [reply src/olympia/promoted/admin.py:20]: I ran your patch locally, but couldn't see where I could maintain approvals. Where does it exist in the admin?

Comment (eviljeff) [reply src/olympia/promoted/admin.py:20]: Inlines are at the bottom of the page, but you can only view the approvals (and delete).  The approval will be done via the reviewer tools.  If you wanted to see this locally you'd do `PromotedApproval.objects.create(version=<version of the addon you've created the PromotedApproval for>, group_id=<a group of your choice, e.g. RECOMMENDED).id)` in a django shell.

Comment (bobsilverberg) [reply src/olympia/promoted/admin.py:20]: Great, thanks @eviljeff!

**Pull request merged**